openc3/lib/openc3/models/plugin_model.rb

```diff
@@ -162,15 +164,14 @@ def self.install_phase2(plugin_hash, scope:, gem_file_path: nil, validate_only:
         gem_path = File.join(temp_dir, "gem")
         FileUtils.mkdir_p(gem_path)
         pkg = Gem::Package.new(gem_file_path)
-        needs_dependencies = pkg.spec.runtime_dependencies.length > 0
         pkg.extract_files(gem_path)
         Dir[File.join(gem_path, '**/screens/*.txt')].each do |filename|
           if File.basename(filename) != File.basename(filename).downcase
             raise "Invalid screen name: #{filename}. Screen names must be lowercase."
           end
         end
-        needs_dependencies = true if Dir.exist?(File.join(gem_path, 'lib'))
-        if needs_dependencies
+        if @needs_dependencies || pkg.spec.runtime_dependencies.length > 0 || Dir.exist?(File.join(gem_path, 'lib'))
```

Comment:
Do a scan here just to look for NEEDS_DEPENDENCIES, and only if it isn't already set.